functests/1_performance/netqueues.go

```diff
@@ -0,0 +1,359 @@
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/profiles"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components"
+)
+
+var _ = Describe("[ref_id: 40307][pao] updating performance profile", func() {
+	var workerRTNodes []corev1.Node
+	var err error
+	var net bool = true
+	var profile *performancev2.PerformanceProfile
+	testutils.BeforeAll(func() {
+		isSNO, err := cluster.IsSingleNode()
+		Expect(err).ToNot(HaveOccurred())
+		RunningOnSingleNode = isSNO
+	})
+	BeforeEach(func() {
+		workerRTNodes, err = nodes.GetByLabels(testutils.NodeSelectorLabels)
+		Expect(err).ToNot(HaveOccurred())
+		workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+		Expect(err).ToNot(HaveOccurred())
+		nodeLabel := testutils.NodeSelectorLabels
+		profile, err = profiles.GetByNodeLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("Updating performance profile for netqueues", func() {
+		It("[test_id:40308][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] Network device queues Should be set to the profile's reserved CPUs count ", func() {
+			devices := make(map[string]int)
+			count := 0
+			if profile.Spec.Net != nil {
+				if profile.Spec.Net.UserLevelNetworking != nil && *profile.Spec.Net.UserLevelNetworking && len(profile.Spec.Net.Devices) == 0 {
+					By("To all non virtual network devices when no devices are specified under profile.Spec.Net.Devices")
+					err := checkDeviceSupport(workerRTNodes, devices)
+					Expect(err).ToNot(HaveOccurred())
+					for _, size := range devices {
+						if size == getReservedCPUSize(profile.Spec.CPU) {
+							count++
+						}
+					}
+					if count == 0 {
+						Skip("Skipping Test: Unable to set Network queue size to reserved cpu count")
+					}
+				}
+			}
+		})
+
+		It("[test_id:40542] Verify the number of network queues of all supported network interfaces are equal to reserved cpus count", func() {
+			devices := make(map[string]int)
+			count := 0
+			if profile.Spec.Net == nil {
+				By("Enable UserLevelNetworking in Profile")
+				profile.Spec.Net = &performancev2.Net{
+					UserLevelNetworking: &net,
+				}
+				profiles.UpdateWithRetry(profile)
+			}
+			// Populate the device map with queue sizes
+			Eventually(func() bool {
+				err := checkDeviceSupport(workerRTNodes, devices)
+				Expect(err).ToNot(HaveOccurred())
+				return true
+			}, cluster.ComputeTestTimeout(200*time.Second, RunningOnSingleNode), testPollInterval*time.Second).Should(BeTrue())
+			for _, size := range devices {
+				if size == getReservedCPUSize(profile.Spec.CPU) {
+					count++
+				}
+			}
+			if count == 0 {
+				Skip("Skipping Test: Unable to set Network queue size to reserved cpu count")
+			}
+			//Verify the tuned profile is created on the worker-cnf nodes:
+			tunedCmd := []string{"tuned-adm", "profile_info", "openshift-node-performance-performance"}
```

Comment:
maybe this check should happen before we compare queue sizes to reserved CPU size ?

Reply:
addressed this in latest patch.

Reply:
@mrniranjan can you point me to the change ?

Reply:
@yanirq https://github.com/mrniranjan/performance-addon-operators/blob/netqueues/functests/1_performance/netqueues.go#L97 the code now first checks the tuned profile and then the size. The same is for all the test cases: